.travis.yml

```diff
@@ -28,68 +119,38 @@ matrix:
+      install: *julia-linux-install
+
+    - &julia-macos
+      os: osx
       env: ARCH="x86_64"
       osx_image: xcode8
-cache:
-  directories:
-    - $TRAVIS_BUILD_DIR/deps/srccache
-    - $TRAVIS_BUILD_DIR/deps/scratch
-    - $TRAVIS_BUILD_DIR/deps/usr-staging
-branches:
-  only:
-    - master
-    - /^release-.*/
-    - /^v\d+\.\d+\.\d+$/
-notifications:
-    email: false
-    irc:
-        channels:
-            - "chat.freenode.net#julia-notifications"
-        on_success: change
-        on_failure: always
-    webhooks:
-        urls:
-          - http://status.julialang.org/put/travis
-          - http://julia.mit.edu:8000/travis-hook
+      install: *julia-macos-install
+
+jobs:
+    include:
+        - <<: *dep-stage
+          <<: *julia-linux-i386
+        - <<: *dep-stage
+          <<: *julia-linux-x86_64
+        - <<: *test-stage
+          <<: *julia-linux-i386
+        - <<: *test-stage
+          <<: *julia-linux-x86_64
+        - <<: *test-stage
+          <<: *julia-macos
+
 before_install:
     - make check-whitespace
-    - if [ `uname` = "Linux" ]; then
-        contrib/travis_fastfail.sh || exit 1;
-        gcc --version;
-        BAR="bar -i 30";
-        BUILDOPTS="-j3 VERBOSE=1 FORCE_ASSERTIONS=1 LLVM_ASSERTIONS=1";
-        echo "override ARCH=$ARCH" >> Make.user;
-        TESTSTORUN="all";
-      elif [ `uname` = "Darwin" ]; then
-        brew update;
-        brew install -v jq pv;
-        BAR="pv -i 30";
-        contrib/travis_fastfail.sh || exit 1;
-        brew tap staticfloat/julia;
-        brew rm --force $(brew deps --HEAD julia);
-        brew install -v --only-dependencies --HEAD julia;
-        brew install -v staticfloat/juliadeps/libgfortran llvm39-julia;
-        BUILDOPTS="-j3 USECLANG=1 LLVM_CONFIG=$(brew --prefix llvm39-julia)/bin/llvm-config LLVM_SIZE=$(brew --prefix llvm39-julia)/bin/llvm-size";
-        BUILDOPTS="$BUILDOPTS VERBOSE=1 USE_BLAS64=0 SUITESPARSE_INC=-I$(brew --prefix suite-sparse-julia)/include FORCE_ASSERTIONS=1";
-        BUILDOPTS="$BUILDOPTS LIBBLAS=-lopenblas LIBBLASNAME=libopenblas LIBLAPACK=-lopenblas LIBLAPACKNAME=libopenblas";
-        for lib in LLVM SUITESPARSE ARPACK BLAS LAPACK GMP MPFR PCRE LIBUNWIND; do
-            BUILDOPTS="$BUILDOPTS USE_SYSTEM_$lib=1";
-        done;
-        export LDFLAGS="-L$(brew --prefix openblas-julia)/lib -L$(brew --prefix suite-sparse-julia)/lib";
-        export DYLD_FALLBACK_LIBRARY_PATH="/usr/local/lib:/lib:/usr/lib:$(brew --prefix openblas-julia)/lib:$(brew --prefix suite-sparse-julia)/lib:$(brew --prefix arpack-julia)/lib";
-        export JULIA_MACOS_SPAWN="DYLD_FALLBACK_LIBRARY_PATH=\"$DYLD_FALLBACK_LIBRARY_PATH\" \$1";
-        export BUILDOPTS="$BUILDOPTS spawn=\$(JULIA_MACOS_SPAWN)";
-        make $BUILDOPTS -C contrib -f repackage_system_suitesparse4.make;
-        TESTSTORUN="all --skip linalg/triangular subarray"; fi # TODO: re enable these if possible without timing out
+    - contrib/travis_fastfail.sh || exit 1;
```

Comment:
apparently I didn't write that script in a very robust way in terms of failing when its dependencies are missing, just gives `contrib/travis_fastfail.sh: line 10: jq: command not found` but didn't cause the build to fail since I think it only tries to run jq inside subshells